lambdas/services/data_collection_service.py

```diff
@@ -75,12 +74,26 @@ def collect_all_data(self) -> list[StatisticData]:
             record_store_data = self.get_record_store_data(
                 dynamodb_scan_result, s3_list_objects_result
             )
-        organisation_data = self.get_organisation_data(dynamodb_scan_result)
 
-        application_data = self.get_application_data()
+        organisation_data = []
+        application_data = []
+
+        for day_start in self._generate_daily_ranges():
+            day_end = day_start + timedelta(days=1)
+            logger.info(f"Collecting data for day: {day_start} to {day_end}")
+
+            organisation_data += self.get_organisation_data(
+                dynamodb_scan_result, day_start, day_end
+            )
+            application_data += self.get_application_data(day_start, day_end)
 
         return record_store_data + organisation_data + application_data
 
+    def _generate_daily_ranges(self):
+        start = datetime.fromtimestamp(self.weekly_collection_start_time)
```

Comment:
If we are converting it back to datetime again, maybe be best to have 'start_time' as a property of this class